frontend/src/components/explorer/gemBrowser/Compartment.vue

```diff
@@ -130,16 +130,14 @@ export default {
     async setup() {
       this.showLoaderMessage = 'Loading compartment data';
       this.cName = this.$route.params.id;
-      if (this.cName !== this.compartment.id) {
-        try {
-          const payload = { model: this.model, id: this.cName };
-          await this.$store.dispatch('compartments/getCompartmentSummary', payload);
-          this.componentNotFound = false;
-          this.showLoaderMessage = '';
-        } catch {
-          this.componentNotFound = true;
-          document.getElementById('search').focus();
-        }
+      try {
+        const payload = { model: this.model, id: this.cName };
+        await this.$store.dispatch('compartments/getCompartmentSummary', payload);
+        this.componentNotFound = false;
+        this.showLoaderMessage = '';
+      } catch {
+        this.componentNotFound = true;
+        document.getElementById('search').focus();
```

Comment:
In principle, I think it makes sense to dispatch a request only if needed, in which the loader message should also be conditional:

```suggestion
      if (this.cName !== this.compartment.id) {
        this.showLoaderMessage = 'Loading compartment data';
        try {
          const payload = { model: this.model, id: this.cName };
          await this.$store.dispatch('compartments/getCompartmentSummary', payload);
          this.componentNotFound = false;
          this.showLoaderMessage = '';
        } catch {
          this.componentNotFound = true;
        }
      }
```

Also, this line is not needed anymore now that the GEM Browser doesn't have its dedicated search bar:
```
          document.getElementById('search').focus();
```

Reply:
Do you remember why that check was added in the first place? I'm wondering if it was necessary due to the `keep-alive` stuff that we used to have. Anyhow, I didn't notice any problems for the other components and none of them had this `if` clause so I simply matched this one to the others.

Reply:
Shouldn't the value of `this.showLoaderMessage` to initialized if `this.cName === this.compartment.id`?

Reply:
The point of the `if` was to avoid making the same request again if the compartment id that is in the component is the same as the id in the url. However, I think Vuex requests are cached so making this request is only a few ms of JS. As @e0 mentioned, this old `if` is likely a relic from `keep-alive`.

@e0 what about the line below, marked for removal in my suggestion?
```
document.getElementById('search').focus();
```

Reply:
@mihai-sysbio Although it's not related to this bug fix I think that absolutely should be removed. The only reason why I don't want to merge it along with this PR is we have the same line in all the other components.

Reply:
The suggested refactor is carried out in #559, so the current PR is self-contained and can be approved.